### ISSUE TITLE
crictl shows the compressed size, and is inconsistent with docker images

### ISSUE THREAD

**Issue by mfranzil** [kubernetes-sigs/cri-tools#1264]
Running `crictl images` (latest version) shows the uncompressed size of each image, which is both misleading and inconsistent with `docker images`.

#### What happened:

Assume I download the latest `nginx` image. Running a simple pod and inspecting the contents of the image cache with `crictl` yields:

```
IMAGE                                  TAG                 IMAGE ID            SIZE
docker.io/library/nginx                latest              f5a6b296b8a29       70.5MB
[...]
```

Downloading the exact image with `docker pull` on the same machine and running `docker images` yields:

```
REPOSITORY   TAG       IMAGE ID       CREATED      SIZE
nginx        latest    f5a6b296b8a2   5 days ago   187MB
```

`crictl` shows the compressed size, while `docker images` shows the uncompressed (on disk) size.

#### What you expected to happen:

Either both tools show the uncompressed size or the compressed size. The latter is probably better (see last paragraph).

#### How to reproduce it (as minimally and precisely as possible):

See the first paragraph.

#### Anything else we need to know?:

Showing the uncompressed size is uninformative and misleading for users trying to debug anything on their system (e.g. low on disk space). It only makes sense when downloading an image, as that is indeed the amount of data that is pulled from the repository.

#### Environment:

- Container runtime or hardware configuration:
  - libvirt-emulated VM via Vagrant with 2 vCPUs and 8GB of RAM, running Kubernetes version v1.27.4
- OS (e.g: `cat /etc/os-release`):
```
NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
- Kernel (e.g. `uname -a`):
```
Linux worker2 5.4.0-146-generic #163-Ubuntu SMP Fri Mar 17 18:26:02 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```
- Others:
  - `crictl version`
```
Version:  0.1.0
RuntimeName:  containerd
RuntimeVersion:  1.6.22
RuntimeApiVersion:  v1
```

**Comment by afbjorklund** [comment]
> crictl version (technically the latest, v1.28.0, but the output of the command is this)

You can use `crictl --version`, if you want the version of the tool instead of the endpoint

**Comment by afbjorklund** [comment]
Seems like it knows both sizes, but chooses to show the blob size for the CRI size.

```
anders@lima-k8s:~$ sudo crictl pull nginx
Image is up to date for sha256:f5a6b296b8a29b4e3d89ffa99e4a86309874ae400e82b3d3993f84e1e3bb0eb9
anders@lima-k8s:~$ sudo crictl images | grep nginx
docker.io/library/nginx                         latest              f5a6b296b8a29       70.5MB
anders@lima-k8s:~$ sudo nerdctl -n k8s.io images nginx
REPOSITORY    TAG       IMAGE ID        CREATED           PLATFORM       SIZE         BLOB SIZE
nginx         latest    6926dd802f40    15 seconds ago    linux/amd64    191.8 MiB    67.2 MiB
```

The CRI API is not so picky, as long as it is a string and an integer it is a name and a size:

https://github.com/kubernetes/cri-api/blob/release-1.28/pkg/apis/runtime/v1/api.proto#L1403

It's "OK" to use either name, and to use either size.

But doesn't seem to be translated at all in `crictl` ?

**Comment by afbjorklund** [comment]
Opening a kubernetes issue to clarify the API is probably the way, then update the runtimes to follow the "new" API

**Comment by SergeyKanzhelev** [comment]
I heard about this inconsistency before, but I don't know the history of this decision. Opening issue as suggested above is definitely way to go.

**Comment by mfranzil** [comment]
> I heard about this inconsistency before, but I don't know the history of this decision. Opening issue as suggested above is definitely way to go.

Yeah done that as kubernetes/kubernetes#120698. Let's see how it goes.

**Comment by afbjorklund** [comment]
The output between `podman images` and `crictl images` is the same, it shows the uncompressed size.

```console
docker@minikube:~$ sudo podman images nginx
REPOSITORY               TAG         IMAGE ID      CREATED      SIZE
docker.io/library/nginx  latest      f5a6b296b8a2  10 days ago  191 MB
docker@minikube:~$ sudo crictl images nginx
IMAGE                     TAG                 IMAGE ID            SIZE
docker.io/library/nginx   latest              f5a6b296b8a29       191MB
docker@minikube:~$ sudo crictl version
Version:  0.1.0
RuntimeName:  cri-o
RuntimeVersion:  1.24.6
RuntimeApiVersion:  v1

```
